featuregates/controller/pkg/feature/feature_controller.go

```diff
@@ -0,0 +1,244 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package feature
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"github.com/vmware-tanzu/tanzu-framework/featuregates/client/pkg/util"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/sets"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	corev1alpha2 "github.com/vmware-tanzu/tanzu-framework/apis/core/v1alpha2"
+)
+
+const contextTimeout = 30 * time.Second
+
+// FeatureReconciler reconciles a Feature object.
+type FeatureReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+// +kubebuilder:rbac:groups=core.tanzu.vmware.com,resources=featuregates,verbs=get;list;watch;update;patch
+// +kubebuilder:rbac:groups=core.tanzu.vmware.com,resources=featuregates/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=core.tanzu.vmware.com,resources=features,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=core.tanzu.vmware.com,resources=features/status,verbs=get;update;patch
+
+// Reconcile reconciles the FeatureGate spec by computing activated, deactivated and unavailable features.
+func (r *FeatureReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	ctxCancel, cancel := context.WithTimeout(ctx, contextTimeout)
+	defer cancel()
+
+	log := r.Log.WithValues("feature", req.NamespacedName)
+	log.Info("Starting reconcile")
```

Comment:
I don't know...if you're looking at pod logs, don't you want to see how many times reconcile was triggered and when? You only really look at pod logs to debug and this is I think the minimum information you want to see when your pod crashes and something goes wrong.

Reply:
i think this message would be useful when debugging the controller and i don't think this would pollute our logs too mych. Also, we do the same thing for [old featuregate controller](https://github.com/vmware-tanzu/tanzu-framework/blob/main/featuregates/controller/pkg/featuregate/featuregate_controller.go#L42) and for [capabilities controller](https://github.com/vmware-tanzu/tanzu-framework/blob/main/capabilities/controller/pkg/capabilities/run/capability_controller.go#L38). If we end up changing this here, we should also make changes in other places for consistency purposes